### ISSUE TITLE
Guidelines for Hacktoberfest contributors

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#66666]
TL;DR: Please **focus on bug fixes** for your first contributions. That's the best way to get familiar with the Godot codebase, and the most useful to help us release Godot 4.0 faster!

---

Hi everyone!

[Hacktoberfest](https://hacktoberfest.com/) is running again and like in previous years, we'd be happy to help new contributors become familiar with contributing to a big Free and Open Source Software project.

Godot is a big project with dozens of PRs submitted everyday even outside of Hacktoberfest, so to ensure a fruitful event for new contributors and maintainers alike, we have some guidelines that we'd like new contributors to respect for this event.

We recently released [Godot 4.0 beta 1](https://godotengine.org/article/dev-snapshot-godot-4-0-beta-1) (and [beta 2](https://godotengine.org/article/dev-snapshot-godot-4-0-beta-2)) and thus Godot's `master` branch is now in **feature freeze**. Our focus is on **fixing reported bugs**, and we ask that new contributors motivated by Hacktoberfest join us in this effort. Our PR reviewers already have a lot to handle, and while they're motivated to help newcomers becoming contributors to Godot, we want to limit this to our ongoing bugfixing effort. This means:

- Hacktoberfest contributions should be fixing a bug. Ideally this should be a **reported bug**, so you can reference it with `Fixes #1234` in the PR description so that it gets closed when the PR is merged.
  * You can browse [reported bugs in the 4.0 milestone](https://github.com/godotengine/godot/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Abug+milestone%3A4.0). You can of course fix a bug you reported yourself, and also fix a bug that may not have been assigned to the 4.0 milestone. Any bugfix is worth having :)

- Feature PRs will not be reviewed in priority and thus may not count as approved for Hacktoberfest. It's best to refrain from working on features altogether at this stage, as they won't be merged before Godot 4.1 in 2023.

- Cosmetic and style PRs such as minor code reorganizations, adding comments, fixing a minor typo, etc. will not be accepted as first contribution. We find that Hacktoberfest contributors often focus on that and this leads to PRs which aren't very useful yet still require a lot of back and forth with maintainers as cosmetic changes are by nature a matter of taste - and newcomers often follow their own taste instead of the established guidelines shared among all contributors.
  * Fixing typos and language issues in the documentation is still very welcome - but consider grouping such fixes together in a single commit/PR so that it's easier to review and merge.

- **Your changes must be tested!** It should compile and work properly.
   * It should go without saying but in 3 days we've had at least 3 PRs that wouldn't compile at all.

Most importantly, the best path to becoming a Godot contributor is to discuss with existing contributors on our [Rocket.Chat instance](https://chat.godotengine.org)/: Join the #new-contributors channel on [chat.godotengine.org](https://chat.godotengine.org)/ and feel free to ask any question about the PR workflow, help to debug and finalize a bugfix commit, etc.

Now, have fun and thanks in advance for your contributions!

**Comment by adamscott** [comment]
🤡🍫👻🧛‍♂️🖱️💻6️⃣6️⃣6️⃣6️⃣6️⃣🖥️⌨️🏰🎃🍬🧟‍♂️

**Comment by PucklaMotzer09** [comment]
How will PRs to godot count towards hacktoberfest. The repo does currently not have the hacktoberfest label? Do I need to specify in my PR that I want it to count?

**Comment by Calinou** [comment]
> How will PRs to godot count towards hacktoberfest. The repo does currently not have the hacktoberfest label? Do I need to specify in my PR that I want it to count?

It's automatic as long as you've registered on the [Hacktoberfest website](https://hacktoberfest.com/).

**Comment by RedMser** [comment]
@Calinou Hacktoberfest is opt-in for repos, according to the [participation guidelines](https://hacktoberfest.com/participation/):

> ## Your PR/MRs must be in a repo tagged with the “hacktoberfest” topic, or be labeled “hacktoberfest-accepted.”
> 
> Hacktoberfest is now opt-in for maintainers, so only contribute to projects that indicate they’re looking for Hacktoberfest PR/MRs.

**Comment by CutestNekoAqua** [comment]
I really like the number of this issue lmao

**Comment by akien-mga** [comment]
Closing as Hacktoberfest has ended. We might reopen this issue for the next round (and maybe repurpose some of these guidelines into general guidelines for new contributors).
